[WIP] Add Node build artifact to build-deps.yml

Blackhex opened the pull request.

eukarpov commented on .github/workflows/build-deps.yml:
```diff
@@ -133,12 +133,28 @@ jobs:
     steps:
 
+    - name: Add 7-Zip to PATH
+      uses: myci-actions/append-to-path-windows@2
```
It might be preinstalled on the runner to avoid usage 3rd party actions.

Blackhex replied:
OK, just note this is not installing it, this is adding it to the PATH, as the installer does not do that automatically. This action was used because adding a dir to PATH in PowerShell was not working. Now, when we are using Batch this should not be needed...

Blackhex replied:
Replaced with
```
echo "PATH=$env:PATH;C:\Program Files\7-Zip" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
```

eukarpov commented on .github/workflows/build-deps.yml:
```diff
@@ -133,12 +133,28 @@ jobs:
         git clone https://github.com/Windows-on-ARM-Experiments/node.git --single-branch -b fix-arm64-compilation --depth 1
         cd node
-        .\vcbuild openssl-no-asm arm64
+        .\vcbuild.bat arm64 package
```
I see, package target was missing

eukarpov commented on .github/workflows/build-deps.yml:
```diff
@@ -178,13 +194,20 @@ jobs:
         New-Item -ItemType Directory -Force -Path emsdk/emscripten
         Expand-Archive -LiteralPath emscripten_package.zip -DestinationPath emsdk\emscripten
         Expand-Archive -LiteralPath llvm-arm64.zip -DestinationPath emsdk
+        Expand-Archive -LiteralPath node-arm64.zip -DestinationPath emsdk\node
```
Build emsdk arm64 was supposed to be used for llvm emscripten package as part of emscripten_build job. I suggest to create separate job for emsdk release which will have dependencies on  emscripten_build and build-nodejs-arm64

Blackhex committed:
Add Node build artifact to build-deps.yml

eukarpov committed:
Fix wasm binaries installation (#11)

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/arcade build 20230203.1 (#288)

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

eukarpov committed:
Add nodejs arm64 (#10)

radekdoulik committed:
Bump emscripten to 3.1.30 (#282)

* [bazel] Set CLOSURE_COMPILER to workaround RBE+symlinks issue (#1037)

* [bazel] Set CLOSURE_COMPILER to workaround RBE+symlinks issue

* space

* specify node_js

* 3.1.10 (#1046)

* Optimize sandbox performance (#1045)

* Optimize sandbox performance

Link just the files needed to compile, link, or archive, rather than the entire directory tree. This drastically improves build times on macOS, where the sandbox is known to be slow (https://github.com/bazelbuild/bazel/issues/8230).

* Linux wants clang to link

* all_files not needed?

* Linux wants llc

* And llvm-ar

* Templated build_file_content

* include node modules glob with linker files. also some minor formatting fixes. (#1052)

* Using bazelisk on macOS CI (#1049)

* Explicit outputs for wasm_cc_binary (#1047)

* Explicit outputs for wasm_cc_binary

* Backwards compatibility

* data_runfiles restore

* restore test_bazel.sh

* Using wrong path on accident

* two separate rules for legacy support

* Added name attribute to wasm_cc_binary rule

* 3.1.11 (#1053)

* 3.1.12 (#1054)

* 3.1.13 (#1055)

* [bazel] Add additional files necessary for building with closure and on RBE (#1057)

* 3.1.14 (#1061)

* 3.1.15 (#1066)

* Pin `latest` to a specific version for arm64-linux (#1065)

Fixes: #1040

* 3.1.16 (#1071)

* 3.1.17 (#1076)

* Exclude msys from path fix function. (#1078)

Fixes: #911

* 3.1.18 (#1081)

* 3.1.18

* Update LLVM include path in Bazel files

* Version 3.1.18-2 (#1083)

3.1.18 had a bad release binary on ARM64 Mac so push an updated version of the release.

* 3.1.19 (#1090)

* Add EMSDK_QUIET to make emsdk_env less chatting (#1091)

Without this the recommended way to silence emsdk_env was to pipe its
stderr to /dev/null.. but then you also loose potentially useful error
message.

Fixes: #946

* 3.1.20 (#1095)

* Add double-quotes to allow spaces in path (#1097)

* 3.1.21 (#1101)

* Update latest-arm64-linux to 3.1.21 (#1102)

* Update XCode version on CircleCI (#1103)

12.2 is being deprecated

* 3.1.22 (#1107)

* 3.1.23 (#1111)

* Avoid exporting EM_CONFIG for modern SDK versions (#1110)

Newer versions of emscipten, starting all the way back in 1.39.13, can
automatically locate the `.emscripten` config file that emsdk creates so
there is no need for the explicit EM_CONFIG environment variable.  Its
redundant and adds unnessary noisce/complexity.

Really, adding emcc to the PATH is all the is needed these days.

One nice thing about this change is that it allows folks to run
whichever emcc they want to and have it just work, even if they have
configured emsdk.   Without this change, if I activate emsdk and I run
`some/other/emcc` then emsdk's `EM_CONFIG` will still be present and
override the configuration embedded in `some/other/emcc`.

e.g. in the same shell, with emsdk activated, I can run both these
commands and have them both just work as expected.

```
$ emcc --version
$ /path/to/my/emcc --version
```

* Use x64 version for Windows on Arm (#1115)

* 3.1.24 (#1122)

* 3.1.25 (#1130)

* [bazel] Switch to platforms-based toolchain resolution (#1036)

* remove "name" attribute from bazel rules (#1131)

* 3.1.26 (#1134)

* Update remote_docker version in CircleCI config (#1117)

20.10.17 is the current default.

* docker image: Change base to Ubuntu 22.04 LTS (jammy) (#1135)

Done to upgrade from CMake 3.16.3 to 3.22.1. CMake 3.21 or newer is needed to build the Qt 6.4.1 sources with emscripten.

Also update to libidn12 to resolve an "Unable to locate package libidn11" error.

* 3.1.27 (#1139)

* Use constants for fixed paths. NFC (#1140)

* Add standalone_wasm feature to bazel emscripten_toolchain (#1145)

* 3.1.28 (#1149)

* Upgrade to rules_nodejs 5.8.0 (#1150)

Fixes https://github.com/emscripten-core/emsdk/issues/1020

* 3.1.29 (#1160)

* Pin Windows CI to Bazel 5.4.0 (#1163)

* Remove reference to fastcomp-latest. NFC (#1164)

fastcomp can only be install using explicit versions names so this name
doesn't work.

* Remove fastcomp SDK and fastcomp build rules. NFC (#1165)

Folks that want to work with fastcomp will now need to use an older
checkout of emsdk.

* 3.1.30 (#1167)

* Bump emscripten to 3.1.30

* Bump clang version

* Do not include test directory

* Update eng/emsdk.proj

Co-authored-by: Alexander Köplinger <alex.koeplinger@outlook.com>

---------

Co-authored-by: Kevin Lubick <kjlubick@users.noreply.github.com>
Co-authored-by: Sam Clegg <sbc@chromium.org>
Co-authored-by: John Firebaugh <john.firebaugh@gmail.com>
Co-authored-by: walkingeyerobot <mitch@thefoley.net>
Co-authored-by: Ezekiel Warren <zaucy@users.noreply.github.com>
Co-authored-by: Heejin Ahn <aheejin@gmail.com>
Co-authored-by: Tim Ebbeke <Tim06TR@gmail.com>
Co-authored-by: Derek Schuff <dschuff@chromium.org>
Co-authored-by: Joel Van Eenwyk <joel.vaneenwyk@gmail.com>
Co-authored-by: Pierrick Bouvier <101587250+pbo-linaro@users.noreply.github.com>
Co-authored-by: Trevor Hickey <TrevorJamesHickey@gmail.com>
Co-authored-by: Fredrik Orderud <forderud@users.noreply.github.com>
Co-authored-by: Robbert van Ginkel <570934+robbertvanginkel@users.noreply.github.com>
Co-authored-by: Alexander Köplinger <alex.koeplinger@outlook.com>

radical committed:
Revert "Bump emscripten to 3.1.30 (#282)" (#291)

This reverts commit f55ecf5518e38508a96afc81e5cfc33bbbc76a8d.

eukarpov committed:
Fix node.js activated path config (#12)

eukarpov committed:
Add pipeline to build and release emsdk arm64 dependencies (#2)

Pipeline to build arm64 version of LLVM, node.js, Binaryen
Package wasm-binaries for arm64 based on LLVM, Binaryen and emscripten builds.

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/arcade build 20230210.3 (#294)

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

Blackhex committed:
Merge remote-tracking branch 'origin/main'

Blackhex committed:
Build Python Arm64 package with update_python.py (#17)

Blackhex committed:
Use released Python Arm64 package (#18)

Blackhex committed:
Add a workflow that runs Emscripten unit tests with the release package (#13)
